### ISSUE TITLE
ovs-ovn has been crashed randomly

### ISSUE THREAD

**Issue by aminmr** [kubeovn/kube-ovn#3196]
# Expected Behavior
All ovs-ovn are working.

# Actual Behavior
The ovs-ovn pod has too many delay errors, but it would not crash, and the state is still running. However, the pods on the specific node could not resolve any IP address in the cluster, including the k8s API server. 

# Additional Info
The `ovs-ovn` pod logs:
```
023-09-09T21:44:28.981Z|176277|reconnect|INFO|tcp:[172.22.22.91]:6642: connecting...
2023-09-09T21:44:28.983Z|176278|reconnect|INFO|tcp:[172.22.22.91]:6642: connected
2023-09-09T21:44:28.994Z|176279|binding|INFO|Claiming lport fluentd-cdxpw.monit for this chassis.
2023-09-09T21:44:28.994Z|176280|binding|INFO|fluentd-cdxpw.monit: Claiming 00:00:00:34:0B:5E 10.16.4.40
2023-09-09T21:44:39.709Z|176281|inc_proc_eng|INFO|node: logical_flow_output, recompute (forced) took 10714ms
2023-09-09T21:44:40.998Z|176282|binding|INFO|Setting lport fluentd-cdxpw.monit up in Southbound
2023-09-09T21:44:40.998Z|176283|lflow_cache|INFO|Detected cache inactivity (last active 36851 ms ago): trimming cache
2023-09-09T21:44:41.020Z|176284|timeval|WARN|Unreasonably long 12031ms poll interval (11587ms user, 440ms system)
2023-09-09T21:44:41.020Z|176285|timeval|WARN|faults: 189971 minor, 0 major
2023-09-09T21:44:41.020Z|176286|timeval|WARN|context switches: 0 voluntary, 76 involuntary
2023-09-09T21:44:41.021Z|176287|poll_loop|INFO|Dropped 8 log messages in last 13 seconds (most recently, 12 seconds ago) due to excessive rate
2023-09-09T21:44:41.021Z|176288|poll_loop|INFO|wakeup due to [POLLIN][POLLERR][POLLHUP] on fd 3 (172.22.22.119:33734<->) at ../lib/stream-fd.c:157 (92% CPU usage)
2023-09-09T21:44:41.021Z|176289|jsonrpc|WARN|tcp:[172.22.22.91]:6642: send error: Broken pipe
2023-09-09T21:44:41.022Z|176290|poll_loop|INFO|wakeup due to 0-ms timeout at tcp:[172.22.22.91]:6642 (92% CPU usage)
2023-09-09T21:44:41.022Z|176291|reconnect|WARN|tcp:[172.22.22.91]:6642: connection dropped (Broken pipe)
2023-09-09T21:44:41.023Z|176292|main|INFO|OVNSB commit failed, force recompute next time.
```
The resource has been allocated for `ovs-ovn` daemonset:
```
        resources:
          limits:
            cpu: "4"
            memory: 6Gi
          requests:
            cpu: "4"
            memory: 6Gi

```
The resource usage of the `ovs-ovn` pod:
![image](https://github.com/kubeovn/kube-ovn/assets/61911987/adafc972-1d01-449e-8191-467bc0aafb4e)

Inside the pod with `top` command:
![image](https://github.com/kubeovn/kube-ovn/assets/61911987/3dec9708-8a46-45ae-8b32-2e12e0dbe392)

`kube-ovn-controller` logs: No suspicious log
`ovn-central` logs: No suspicious log

**Note: By restarting the ovs-ovn pod, the problem will be fixed! But why it has occurred?!**

I tried to figure out my problem via issue #3051 but It was not helpful.

- Kubernetes version:
`1.25.6`

  **Output of `kubectl version`:**

```  
Client Version: version.Info{Major:"1", Minor:"27", GitVersion:"v1.27.4", GitCommit:"fa3d7990104d7c1f16943a67f11b154b71f6a132", GitTreeState:"archive", BuildDate:"2023-07-20T07:37:53Z", GoVersion:"go1.20.6", Compiler:"gc", Platform:"linux/amd64"}
Kustomize Version: v5.0.1
Server Version: version.Info{Major:"1", Minor:"25", GitVersion:"v1.25.6", GitCommit:"ff2c119726cc1f8926fb0585c74b25921e866a28", GitTreeState:"clean", BuildDate:"2023-01-18T19:15:26Z", GoVersion:"go1.19.5", Compiler:"gc", Platform:"linux/amd64"}
```
- **kube-ovn version**:
`1.12.0`

- operation-system/kernel version:
  
  **Output of `awk -F '=' '/PRETTY_NAME/ { print $2 }' /etc/os-release`:**
```
"Debian GNU/Linux 11 (bullseye)"
```
  **Output of `uname -r`:**
```
5.10.0-25-amd64
```

**Comment by aminmr** [comment]
For further information, this occurred on multiple cluster nodes, and the ovs-ovn restarting is not solving the problem anymore. I must drain the node and reboot the host to fix this problem. I hope someone can determine what the problem is here.

**Comment by oilbeater** [comment]
> However, the pods on the specific node could not resolve any IP address in the cluster, including the k8s API server.

Hi @aminmr, are all the pods on the node unable to resolve the IP, or is it only some of the pods that cannot resolve the IP?And did you test other network connectivity like ping other Pod IP directly. 

I see the CPU and memory usage for ovn-controller and ovs-vswitchd are high so I guess it may related to the scale. At the moment we only know the subnet overlay to underlay interconnection may lead to OOM. Can you list the features that you used, so we can further investigate which features may lead to the resource usage.

**Comment by aminmr** [comment]
Hi @oilbeater 
Yes. I scheduled a nginx pod on the specific node with the nodeName flag, and the nginx pod also has the problem.
The first pods that show signs of network issues are  Fluentd and black-box exporter pods, which couldn't connect to the API server and generate this log:
```
2023-09-09 13:22:43 +0000 [error]: config error file="/fluentd/etc/../../etc/fluent/fluent.conf" error_class=Fluent::ConfigError error="Invalid Kubernetes API v1 endpoint https://10.96.0.1:443/api: Timed out connecting to server"
```
When this problem continues in some nodes the kube-ovn-pinger pods are failed to and this is the log:
```
E0911 00:14:12.237761  724490 config.go:160] failed to get self pod kube-system/kube-ovn-pinger-8dnb5: Get "https://10.96.0.1:443/api/v1/namespaces/kube-system/pods/kube-ovn-pinger-8dnb5?timeout=15s": net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)
```

As you said about the resource usage of ovn-controller, I already checked the ovn-controller, but the resource usage is not too high:(24 Hours resource usage of ovn-controller)
![image](https://github.com/kubeovn/kube-ovn/assets/61911987/cc886265-8393-4c87-9843-35f4d27d73b4)

But about ovs-vswitched, how can I measure the resource usage? Is this component inside the ovs-ovn pod? Why the total usage is not too high?

I use the basic feature of kube-ovn. If you are interested in specific features and how exactly to list these features, let me know to provide this information for you.
